TESTS/mbed_drivers/flashiap/main.cpp

```diff
@@ -178,15 +178,103 @@ void flashiap_program_error_test()
     TEST_ASSERT_EQUAL_INT32(0, ret);
 }
 
+void flashiap_timing_test()
+{
+    FlashIAP flash_device;
+    uint32_t ret = flash_device.init();
+    TEST_ASSERT_EQUAL_INT32(0, ret);
+    mbed::Timer timer;
+    int num_write_sizes;
+    unsigned int curr_time, byte_usec_ratio;
+    unsigned int avg_erase_time = 0;
+    unsigned int max_erase_time = 0, min_erase_time = (unsigned int) -1;
+    const unsigned int max_writes = 128;
+    const unsigned int max_write_sizes = 6;
+    const unsigned int max_byte_usec_ratio = 200;
+    uint32_t sector_size;
+
+    uint32_t page_size = flash_device.get_page_size();
+    uint32_t write_size = page_size;
+
+    uint32_t address = flash_device.get_flash_start();
+    uint32_t end_address = address + flash_device.get_flash_size();
+    printf("Flash layout:\n");
+    while(address < end_address) {
+        sector_size = flash_device.get_sector_size(address);
+        printf("Address 0x%08x, size %6d\n", address, sector_size);
```

Comment:
can we reduce the number of print for the test, we may hit similar issue to IOTMORF-1861